components/my-table-filter/my-table-filter.js

```diff
@@ -0,0 +1,152 @@
+import Grid from '@mui/material/Grid'
+import Button from '@mui/material/Button'
+import FormControl from '@mui/material/FormControl'
+import InputLabel from '@mui/material/InputLabel'
+import Select from '@mui/material/Select'
+import MenuItem from '@mui/material/MenuItem'
+import TextField from '@mui/material/TextField'
+import PropTypes from 'prop-types'
+import { useFormik } from 'formik'
+
+function MyTableFilter({ onSubmit }) {
+  const formik = useFormik({
+    initialValues: {
+      status: '',
+      type: '',
+      number: '',
+      client_name: '',
+      date: '',
+      total_w_vat: '',
+    },
+    onSubmit: (data) => onSubmit(data),
+  })
```

Comment:
Because I think using formik is easier to maintain the code and follow the pattern of lib and validate fields. Make my own form code in some cases is ok, but if a large team needs to maintain the form code this maybe will be a problem, because in most cases every developer does it in a different way, and using a lib these things do not happens.